src/Razor/src/Microsoft.VisualStudio.Editor.Razor/Settings/ClientSettingsManager.cs

```diff
@@ -6,33 +6,36 @@
 using System.ComponentModel.Composition;
 using Microsoft.CodeAnalysis.ExternalAccess.Razor;
 using Microsoft.CodeAnalysis.Razor.Settings;
+using Microsoft.Extensions.Logging;
 using Microsoft.VisualStudio.Threading;
 
 namespace Microsoft.VisualStudio.Editor.Razor.Settings;
 
 [Export(typeof(IClientSettingsManager))]
-internal class ClientSettingsManager : IClientSettingsManager
+internal class ClientSettingsManager : IClientSettingsManager, IDisposable
 {
     public event EventHandler<ClientSettingsChangedEventArgs>? ClientSettingsChanged;
+    public event EventHandler<bool>? FeedbackRecordingChanged;
 
     private readonly object _settingsUpdateLock = new();
+
     private readonly IAdvancedSettingsStorage? _advancedSettingsStorage;
     private readonly RazorGlobalOptions? _globalOptions;
-    private ClientSettings _settings;
+    private FeedbackRecordingWatcher? _feedbackRecordingWatcher = new();
```

Comment:
Would it be clearer to have this be a MEF service, so that the in memory logger can consume it rather than the "Client settings manager" which it doesn't really care about.